pytorch_lightning/trainer/supporters.py

```diff
@@ -319,10 +318,13 @@ def __len__(self) -> int:
         return self._calc_num_data(self.datasets, self.mode)
 
 
+class DataLoaderDict(Dict):
```

Comment:
Can you add a docstring indicating why this is necessary?

Reply:
this is only to allow type checks. I'll add a comment.